cmd/ctl/pkg/status/certificate/types.go

```diff
@@ -442,3 +432,63 @@ func (orderStatus *OrderStatus) String() string {
 
 	return output
 }
+
+func CertificateStatusDeepEqual(x, y *CertificateStatus) bool {
+	if x == nil || y == nil {
+		return x == y
+	}
+
+	return reflect.DeepEqual(x.Name, y.Name) && reflect.DeepEqual(x.Namespace, y.Namespace) &&
+		reflect.DeepEqual(x.CreationTime, y.CreationTime) && reflect.DeepEqual(x.Conditions, y.Conditions) &&
+		reflect.DeepEqual(x.DNSNames, y.DNSNames) && reflect.DeepEqual(x.Events, y.Events) &&
+		reflect.DeepEqual(x.NotBefore, y.NotBefore) && reflect.DeepEqual(x.NotAfter, y.NotAfter) &&
+		reflect.DeepEqual(x.RenewalTime, y.RenewalTime) &&
+		IssuerStatusDeepEqual(x.IssuerStatus, y.IssuerStatus) &&
+		SecretStatusDeepEqual(x.SecretStatus, y.SecretStatus) &&
+		CRStatusDeepEqual(x.CRStatus, y.CRStatus) &&
+		OrderStatusDeepEqual(x.OrderStatus, y.OrderStatus)
+
+}
+
+func IssuerStatusDeepEqual(x, y *IssuerStatus) bool {
```

Comment:
oh wow... okay could we add a note why we did this?
Also is there something in some package like https://godoc.org/github.com/stretchr/testify/assert that can help us here?